packages/core/src/geometry/Buffer.ts

```diff
@@ -107,6 +115,20 @@ export class Buffer
         this.data = null;
     }
 
+    /**
+     * setting index to true, will set the buffer type to be an ELEMENT_ARRAY_BUFFER
+     * false will set its type to ARRAY_BUFFER
+     * for backwards compatibility
```

Comment:
```suggestion
     * Flags whether this is an index buffer.
     *
     * Index buffers are of type `ELEMENT_ARRAY_BUFFER`. Note that setting this property to false will make
     * the buffer of type `ARRAY_BUFFER`. 
     * 
     * For backwards compatibility.
```

I suggest we also remove the "backwards compat" comment unless we actually plan to deprecate the setter in the future. Having a `index` property seems pretty valid.